phoenix-core/src/main/java/org/apache/phoenix/util/ExpressionUtil.java

```diff
@@ -68,4 +74,85 @@ public static boolean isPkPositionChanging(TableRef tableRef, List<Expression> p
         return false;
     }
 
+    public static boolean isColumnConstant(Expression columnExpression, Expression whereExpression) {
+        if(whereExpression == null) {
+            return false;
+        }
+        IsColumnConstantExpressionVisitor isColumnConstantExpressionVisitor =
+                new IsColumnConstantExpressionVisitor(columnExpression);
+        whereExpression.accept(isColumnConstantExpressionVisitor);
+        return isColumnConstantExpressionVisitor.isConstant();
+    }
+
+    private static class IsColumnConstantExpressionVisitor extends StatelessTraverseNoExpressionVisitor<Void> {
```

Comment:
I'm hoping we don't need a new visitor here. There are probably other cases to check for besides ComparisonExpression. For example, InListExpression, maybe CoerceExpression, etc.

Reply:
I did not find a existing visitor  which can statisfy the requirement , I would check more Expression besides ComparisonExpression.